### PR TITLE
chore: Refactors sidebar components used in help center

### PR DESCRIPTION
**Pull request by nithindavid**
# Pull Request Template

## Description

Please include a summary of the change and issue(s) fixed. Also, mention relevant motivation, context, and any dependencies that this change requires.
Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/javascript/dashboard/routes/dashboard/helpcenter/pages/portals/PortalDetails.vue [hunk 1]
```diff
@@ -37,24 +37,31 @@ export default {
...
         this.alertMessage = this.$t(
           'HELP_CENTER.PORTAL.ADD.API.SUCCESS_MESSAGE_FOR_BASIC'
         );
+
+        await this.$store.dispatch('portals/index').then(() => {
+          this.$store.dispatch('categories/index', {
+            portalSlug: this.selectedPortalSlug,
+          });
+        });
+
```
Comment (pranavrajs) [inline app/javascript/dashboard/routes/dashboard/helpcenter/pages/portals/PortalDetails.vue:54]:
```suggestion
```

File: app/javascript/dashboard/routes/dashboard/helpcenter/pages/portals/PortalDetails.vue [hunk 2]
```diff
@@ -6,7 +6,7 @@
         'HELP_CENTER.PORTAL.ADD.CREATE_FLOW_PAGE.BASIC_SETTINGS_PAGE.CREATE_BASIC_SETTING_BUTTON'
       )
     "
-    @submit="updateBasicSettings"
+    @submit="createBasicPortal"
```
Comment (pranavrajs) [inline app/javascript/dashboard/routes/dashboard/helpcenter/pages/portals/PortalDetails.vue:9]:
```suggestion
    @submit="createPortal"
```

File: app/javascript/dashboard/routes/dashboard/helpcenter/pages/portals/PortalDetails.vue [hunk 3]
```diff
@@ -37,24 +37,24 @@ export default {
   },
 
   methods: {
-    async updateBasicSettings(portal) {
+    async createBasicPortal(portal) {
```
Comment (pranavrajs) [inline app/javascript/dashboard/routes/dashboard/helpcenter/pages/portals/PortalDetails.vue:40]:
```suggestion
    async createPortal(portal) {
```

File: app/javascript/dashboard/components/layout/sidebarComponents/SecondaryNavItem.vue [hunk 4]
```diff
@@ -161,8 +145,8 @@ export default {
         this.menuItem.toStateName === 'settings_applications'
       );
     },
-    isArticlesView() {
-      return this.$store.state.route.name === this.menuItem.toStateName;
+    isCurrentRoute() {
```
Comment (iamsivin) [inline app/javascript/dashboard/components/layout/sidebarComponents/SecondaryNavItem.vue:148]: I tested it locally and LGTM.

@nithindavid  Active colour for the count is not working.

**Before**
image

**After**
image

**Pull request merged**